### PR TITLE
MAINT: revert __skip_array_function__ from NEP-18

### PR DESCRIPTION
**Pull request by shoyer**
xref GH-13624, GH-12028

~~TODO: update tests/CI for NUMPY_EXPERIMENTAL_ARRAY_FUNCTION=0~~

### CODE REVIEW

**Review by mhvk — APPROVED** [review]
This also looks good. I wondered a bit about even defining `_implementation` when things are not enabled, but in the end it is probably best to have minimal differences.

**Comment by shoyer** [comment]
> I wondered a bit about even defining _implementation when things are not enabled, but in the end it is probably best to have minimal differences.

I think we will need this for a few functions that access the implementation internally... we'll see when I run tests with `NUMPY_EXPERIMENTAL_ARRAY_FUNCTION=0`

**Comment by shoyer** [comment]
> I wondered a bit about even defining `_implementation` when things are not enabled, but in the end it is probably best to have minimal differences.

This actually doesn't work if the function is written in C, so I've removed it (that's why you write tests, I suppose!)

File: .travis.yml [hunk 1]
```diff
@@ -64,6 +64,8 @@ matrix:
       env: NPY_RELAXED_STRIDES_CHECKING=0
     - python: 3.6
       env: USE_WHEEL=1 NPY_RELAXED_STRIDES_DEBUG=1
+    - python: 3.6
+      env: NUMPY_EXPERIMENTAL_ARRAY_FUNCTION=0
```
Comment (shoyer) [inline .travis.yml]: Should we consolidate this CI run into one of the others?

Comment (charris) [reply .travis.yml]: We get 5 parallel runs and we are already over 10, so as long as there are fewer than 15 test runs it shouldn't make much difference. The runs vary in time, so it isn't possible to predict exactly what will happen, but close enough.

**Comment by shoyer** [comment]
OK, tests with array function disabled are passing now.

**Pull request merged**